src/wallet/rpcwallet.cpp

```diff
@@ -2326,116 +1901,43 @@ void ListTransactions(const CWalletTx& wtx, const std::string& strAccount, int n
     int depth = wtx.GetDepthInMainChain();
     if (listReceived.size() > 0 && depth >= nMinDepth) {
         for (const COutputEntry& r : listReceived) {
-            std::string account;
+            std::string label;
             if (pwalletMain->HasAddressBook(r.destination))
-                account = pwalletMain->GetNameForAddressBookEntry(r.destination);
-            if (fAllAccounts || (account == strAccount)) {
-                UniValue entry(UniValue::VOBJ);
-                if (involvesWatchonly || (::IsMine(*pwalletMain, r.destination) & ISMINE_WATCH_ONLY))
-                    entry.pushKV("involvesWatchonly", true);
-                if (IsDeprecatedRPCEnabled("accounts")) entry.pushKV("account", account);
-                MaybePushAddress(entry, r.destination);
-                if (wtx.IsCoinBase()) {
-                    if (depth < 1)
-                        entry.pushKV("category", "orphan");
-                    else if (wtx.GetBlocksToMaturity() > 0)
-                        entry.pushKV("category", "immature");
-                    else
-                        entry.pushKV("category", "generate");
-                } else {
-                    entry.pushKV("category", "receive");
-                }
-                entry.pushKV("amount", ValueFromAmount(r.amount));
-                if (pwalletMain->HasAddressBook(r.destination)) {
-                    entry.pushKV("label", account);
-                }
-                entry.pushKV("vout", r.vout);
-                if (fLong)
-                    WalletTxToJSON(wtx, entry);
-                ret.push_back(entry);
+                label = pwalletMain->GetNameForAddressBookEntry(r.destination);
+            UniValue entry(UniValue::VOBJ);
+            if (involvesWatchonly || (::IsMine(*pwalletMain, r.destination) & ISMINE_WATCH_ONLY))
+                entry.pushKV("involvesWatchonly", true);
+            MaybePushAddress(entry, r.destination);
+            if (wtx.IsCoinBase()) {
```

Comment:
Not for this PR but this should be `if (wtx.IsCoinBase() || wtx.IsCoinstake()) {`